lib/active_versioning/model/versioned.rb

```diff
@@ -0,0 +1,87 @@
+module ActiveVersioning
+  module Model
+    module Versioned
+      RecordNotPersisted = Class.new(StandardError)
+
+      mattr_accessor :blacklisted_attributes do
+        %w(created_at updated_at published)
+      end
+
+      def self.included(base)
+        base.class_eval do
+          has_many :versions, -> { newest_first }, as: :versionable, dependent: :destroy
+
+          after_create :create_version!
+
+          # Set by controller params
+          attr_accessor :version_author
+        end
+      end
+
+      def live?
+        true
+      end
+
+      def version?
+        false
+      end
+
+      def current_draft(force_reload = false)
+        unless persisted?
+          raise RecordNotPersisted.new("#{self} must be persisted to create a draft version")
+        end
+
+        @current_draft = nil if force_reload
+
+        @current_draft ||= begin
+          # FIXME: Gracefully handle NoMethodError and UnknownAttributeError
+          # If past versions contain renamed/different attributes, this causes problems
+          # Detect such attributes and warn appropriately.
+          # Create a nice errors object that encapsulates this information that can be used to diff.
+
+          VersionProxy.new(versions.draft.first_or_create(
+            object: versioned_attributes,
+            event:  'draft'
```

Comment:
YES.  I'm definitely going to do this.  Probably need to add some validations around the acceptable events as well.  :+1: